### PR TITLE
feat: add base deployments

### PR DESCRIPTION
**Pull request by md0x**
Changes proposed in this PR:
- Add Base network as a supported network
- Deploy Base contracts and L1 Parent Messenger
- Verify contracts
- Run all the required L1 and L2 configurations
- Update configuration and verification scripts
- Update readmes

See the [comment below](https://github.com/UMAprotocol/protocol/pull/4729#issuecomment-2009343742) with the required remaining governance action to finalise the Base setup

### CODE REVIEW

**Comment by md0x** [comment]
This still requires a governance action to:
- Set the messenger in the [OracleHub](https://etherscan.io/address/0x8fE658AeB8d55fd1F3E157Ff8B316E232ffFF372#writeContract) for the new chain 8453
- Set the messenger in the [GovernorHub](https://etherscan.io/address/0x94520d90A4EBaA98e5A7B8D6809463f65198C104#writeContract) for the new chain 8453

🌕 Verifying L1 contract state 🌕
  GovernorHub
    - Owner set to Governor: ✅
    - Messenger for chain ID 8453 set to ParentMessenger: ❌
  OracleHub
    - Owner set to Governor: ✅
    - Messenger for chain ID 8453 set to ParentMessenger: ❌
  Base_ParentMessenger
    - Owner set to Governor: ✅
    - Child chain ID set to 8453: ✅
    - Set childMessenger address: ✅
    - Set oracleHub address: ✅
    - Set governorHub address: ✅
    - Set oracleSpoke address: ✅
    - Set governorSpoke address: ✅
  Registry
    - OracleHub registered: ✅

🌚 Verifying L2 contract state 🌚
  Registry
    - OptimisticOracle registered: ✅
    - OptimisticOracleV2 registered: ✅
    - OptimisticOracleV3 registered: ✅
    - Owned by GovernorSpoke: ✅
  Store
    - Owned by GovernorSpoke: ✅
  IdentifierWhitelist
    - Owned by GovernorSpoke: ✅
  OptimisticOracleV3
    - Owned by GovernorSpoke: ✅
  AddressWhitelist
    - Owned by GovernorSpoke: ✅
  Base_ChildMessenger
    - Set oracleSpoke: ✅
    - Set parentMessenger: ✅
  Finder
    - Owned by GovernorSpoke: ✅
    - Set "Store" in Finder: ✅
    - Set "IdentifierWhitelist": ✅
    - Set "CollateralWhitelist" in Finder: ✅
    - Set "Oracle" in Finder (to OracleSpoke): ✅
    - Set "OptimisticOracle" in Finder: ✅
    - Set "OptimisticOracleV2" in Finder: ✅
    - Set "OptimisticOracleV3" in Finder: ✅
    - Set "ChildMessenger" in Finder: ✅

**Comment by Reinis-FRP** [comment]
Might also need to add gas estimator settings in `packages/financial-templates-lib/src/helpers/GasEstimator.ts`

File: packages/core/contracts/cross-chain-oracle/README.md [hunk 1]
```diff
@@ -70,24 +71,26 @@ yarn hardhat --network arbitrum etherscan-verify --api-key <ETHERSCAN_KEY> --lic
...
 
 6. Setup l2 contracts
 
 ```sh
-yarn hardhat setup-l2-xchain --network [arbitrum/boba/optimism]
+yarn hardhat setup-l2-xchain --network [arbitrum/boba/optimism/base]
 ```
 
-7. At this point, the cross chain contract suite setup is complete. We will now deploy the `OptimisticOracle` and required contracts to the L2.
+7. At this point, the cross chain contract suite setup is complete. We will now deploy the `OptimisticOracle,OptimisticOracleV2,OptimisticOracleV3` and required contracts to the L2.
```
Comment (Reinis-FRP) [inline packages/core/contracts/cross-chain-oracle/README.md:90]: doesn't OptimisticOracleV3 depend on first having finder and whitelist setups as in later stages

File: packages/core/networks/8453.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    "address": "0x1037A21a30aEfF90c269b01c3933eB9a5285d9b8"
+  },
+  {
+    "contractName": "Store",
+    "address": "0xa799650614f84e0B7E7E99e12d5DFc241907cd85"
+  },
+  {
+    "contractName": "OptimisticOracleV3",
+    "address": "0x2aBf1Bd76655de80eDB3086114315Eec75AF500c"
+  }
```
Comment (Reinis-FRP) [inline packages/core/networks/8453.json:46]: can you also add OptimisticGovernor using `packages/core/deploy/058_deploy_optimistic_governor.js`

**Review by daywiss — APPROVED** [review]
lgtm

**Pull request merged**